x/sync/sync_test.go

```diff
@@ -597,34 +598,26 @@ func TestFindNextKeyRandom(t *testing.T) {
 		slices.SortFunc(localKeyIDs, serializedPathLess)
 
 		// Filter out keys that are before the last received key
-		findBounds := func(keyIDs []keyAndID) (int, int) {
-			var (
-				firstIdxInRange      = len(keyIDs)
-				firstIdxInRangeFound = false
-				firstIdxOutOfRange   = len(keyIDs)
-			)
+		findBounds := func(keyIDs []keyAndID) int {
+			firstIdxAfterLastReceived := len(keyIDs)
 			for i, keyID := range keyIDs {
-				if !firstIdxInRangeFound && bytes.Compare(keyID.key.Value, lastReceivedKey) >= 0 {
-					firstIdxInRange = i
-					firstIdxInRangeFound = true
-					continue
-				}
-				if bytes.Compare(keyID.key.Value, rangeEnd) > 0 {
-					firstIdxOutOfRange = i
+				if bytes.Compare(keyID.key.Value, lastReceivedKey) > 0 {
+					firstIdxAfterLastReceived = i
 					break
 				}
+
 			}
-			return firstIdxInRange, firstIdxOutOfRange
+			return firstIdxAfterLastReceived
```

Comment:
Returning the last index wasn't necessary so I removed it. Now the test is ever so slightly slower but more readable.